modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/TypeDefinition.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.config.spring.dsl.processor;
+
+/**
+ * Defines the actual {@code Class} for the domain object to be created.
+ */
+public class TypeDefinition
+{
+    private Class<?> type;
```

Comment:
Can't be final since both are created by a builder. There's no way to change them anyway.